### PR TITLE
Add new safemode rule category for ignoring sounds

### LINKED ISSUE #28672: Extremely annoying (I)gnore, and how it could be fixed
**Is your feature request related to a problem? Please describe.**  
While (for example) moving 100 two-by-fours from one space to another using the advanced inventory, if there's any kind of loud noise, player is stuck having to hold the I key to ignore the sounds. I'm pretty sure everyone is already familiar with and have encountered the problem at least one time in the game.

**Describe the solution you'd like**  
IMHO, we should have something like "ignore sounds entirely" option, or we should use the safe mode to say that we don't want safety while moving these items. There is a problem though, and that's: What happens if a zombie comes and kills me while i'm moving the stuff and have the safe mode off?
Well, I think safe mode works as a "sight safe mode" currently, so adding sound to that would be weird and hard. Maybe it's better to add an additional safe mode called "sound safe mode" which controls safety for sound only. Also if you decide to add scent detection to the player some time in the future (Like, for example, "you can smell rotten meat here!"), you could add another safe mode for scents called "scent safe mode" which would control the scent detection only. With more safe modes, every player has more in-game control in my opinion.

**Additional context**  
For me, this is a must-do "fix". I've seen people getting really mad about having to press 'I' to ignore all the messages, and i've experienced it myself too. I think it's super annoying, and with more safe modes and more control, this could be fixed nicely.

Thank you for your time!

### LINKED ISSUE #36151: Make interrupt messages being an option to turn on/off
**Is your feature request related to a problem? Please describe.**  
A problem, well, when you have auto-foraging turned on, and you're getting 

https://media.discordapp.net/attachments/252368987210514433/656078283619958784/aaa.png
https://media.discordapp.net/attachments/252368987210514433/656078291903447051/bbb.png

every step you make, you tell me. If I may be so blunt: It's fucking annoying! I need it to go away! I like living on the edge! If I die, it's my fault. Not the games. No handholding for me.

Also same stuff when you have 1-shot weapons. You reload, message "blabla close", you aim, message "blabla close", you shot, you reload, message, you aim, message, you get the picture. That's not my definition of fun. BEing interrupted at every other turn.

Please make these interrupt messages an option to be turned off / on by the user! "Discussion"/Feedback going here: https://discordapp.com/channels/252365627543519232/252368987210514433/656078284164956170

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
SUMMARY: Interface "Adds new safemode rule category for ignoring sounds"

#### Purpose of change
Fixes #28672
Fixes #36151
In the safe mode manager adds a new rule category: sounds to allow ignoring sounds.  

#### Describe the solution
Added categories to safe mode rules with at present only two but future plans to merge auto pickup in as a third.  Hostile category is unchanged existing functionality and is default.  The new sound category allows the same pattern matching search combined with distance.  Any sound matching a rule and at or further away  than the distance will be auto ignored if the rule is a whitelist, if it's a blacklist it will ever be auto ignored..

#### Describe alternatives you've considered
#35436 and #35983

### CODE REVIEW

File: src/safemode_ui.cpp [hunk 1]
```diff
@@ -128,6 +130,7 @@ void safemode::show( const std::string &custom_name_in, bool is_safemode_in )
     mvwprintz( w_header, point( column_pos[COLUMN_ATTITUDE] + 2, 3 ), c_white, _( "Attitude" ) );
     mvwprintz( w_header, point( column_pos[COLUMN_PROXIMITY] + 2, 3 ), c_white, _( "Dist" ) );
     mvwprintz( w_header, point( column_pos[COLUMN_WHITE_BLACKLIST] + 2, 3 ), c_white, _( "B/W" ) );
+    mvwprintz( w_header, point( column_pos[COLUMN_CATEGORY] + 2, 3 ), c_white, _( "Cat" ) );
```
Comment (Qrox) [inline src/safemode_ui.cpp]:
```suggestion
    mvwprintz( w_header, point( column_pos[COLUMN_CATEGORY] + 2, 3 ), c_white, pgettext( "category", "Cat" ) );
```
You don't want it translate to "feline" in other languages.

**Comment by OzoneH3** [comment]
* ~~I can't get this to work with single sounds (eg footsteps), it seems to only work with *.~~
It works with "footsteps." and "whump!" which both add something after the name, another reason to implement the next point.

* With creatures you can press ~ to create a new rule to ignore them. Would be very useful for Sounds as well.

* The player should be able to Black/Whitelist sounds as well.

* Needs a different Example prompt when changing the rule in category SOUND.

* Maybe consider renaming the categories to HOSTILES and SOUNDS, its kinda confusing with Creature:HOSTILE also being a thing.

* While in Category Sound, pressing Return on B/W "---" it changes the Dist from 0 to ---.

**Comment by Ramza13** [comment]
Just updated it based on OzoneH3's feedback.  Right now whitelisted sounds will be auto ignored while blacklisted ones will never be auto ignored.  Not sure if it makes more sense this way or reversed.

**Comment by ZhilkinSerg** [comment]
Needs conflict resolution.

**Pull request merged**